### ISSUE TITLE
[YSQL] Query execution with NLJ and IndexScan now takes long time 

### ISSUE THREAD

**Issue by qvad** [yugabyte/yugabyte-db#15397]
Jira Link: [DB-4528](https://yugabyte.atlassian.net/browse/DB-4528)
### Description

Current master behaviour 2.17.1.0-b351:
```
taqo=# explain analyze SELECT t1.k1, t1.k2, t2.v1, t2.v2 FROM t1 JOIN t2 on t1.k1 = t2.k1 WHERE t1.k1 >= 2500 AND t1.k1 < 25100 AND t2.k1 >= 2500 AND t2.k1 < 25100 GROUP BY t1.k1, t1.k2, t2.v1, t2.v2;
                                                               QUERY PLAN
----------------------------------------------------------------------------------------------------------------------------------------
 Group  (cost=8.04..8.06 rows=1 width=72) (actual time=2031000.139..2031005.593 rows=22600 loops=1)
   Group Key: t1.k1, t1.k2, t2.v1, t2.v2
   ->  Sort  (cost=8.04..8.05 rows=1 width=72) (actual time=2031000.136..2031002.025 rows=22600 loops=1)
         Sort Key: t1.k1, t1.k2, t2.v1, t2.v2
         Sort Method: external merge  Disk: 3568kB
         ->  Nested Loop  (cost=0.00..8.03 rows=1 width=72) (actual time=8.991..2030921.752 rows=22600 loops=1)
               Join Filter: (t1.k1 = t2.k1)
               Rows Removed by Join Filter: 510737400
               ->  Index Scan using t1_pkey on t1  (cost=0.00..4.01 rows=1 width=36) (actual time=4.118..53.790 rows=22600 loops=1)
                     Index Cond: ((k1 >= 2500) AND (k1 < 25100))
               ->  Index Scan using t2_pkey on t2  (cost=0.00..4.01 rows=1 width=40) (actual time=4.019..88.240 rows=22600 loops=22600)
                     Index Cond: ((k1 >= 2500) AND (k1 < 25100))
 Planning Time: 11.267 ms
 Execution Time: 2031007.848 ms
 Peak Memory Usage: 4947 kB
(15 rows)
```

Previous master behaviour with CBO:
```
Group  (cost=1374.07..1374.09 rows=1 width=72) (actual time=3694.170..3699.639 rows=22600 loops=1)
  Group Key: t1.k1, t1.k2, t2.v1, t2.v2
  ->  Sort  (cost=1374.07..1374.08 rows=1 width=72) (actual time=3694.168..3696.056 rows=22600 loops=1)
        Sort Key: t1.k1, t1.k2, t2.v1, t2.v2
        Sort Method: external merge  Disk: 3568kB
        ->  Nested Loop  (cost=0.00..1374.06 rows=1 width=72) (actual time=3.496..3675.485 rows=22600 loops=1)
              ->  Index Scan using t2_pkey on t2  (cost=0.00..1149.59 rows=9962 width=40) (actual time=3.261..17.742 rows=22600 loops=1)
                    Index Cond: ((k1 >= 2500) AND (k1 < 25100))
              ->  Index Scan using t1_pkey on t1  (cost=0.00..0.01 rows=1 width=36) (actual time=0.154..0.155 rows=1 loops=22600)
                    Index Cond: ((k1 = t2.k1) AND (k1 >= 2500) AND (k1 < 25100))
Planning Time: 0.995 ms
Execution Time: 3701.075 ms
Peak Memory Usage: 4942 kB
```

DDL: 
[test.create.sql.zip](https://github.com/yugabyte/yugabyte-db/files/10277355/test.create.sql.zip)

**Comment by gauravk-in** [comment]
The bug in #15094 was shadowing this bug in selectivity estimation.

The root cause for this issue is in histogram selectivity estimation. 

In this test, the columns k1 and k2 in both tables have all values in range [1, 500000]. Yugabyte collects histogram statistics for each column which looks like [154982, 360946, 381242, ... , 499394, 499698, 500000]. The query has a filter for k1 in the range [2500, 25100). Note that this range falls below the lowest value in the histogram. For the filter k1 < 25100, the histogram selectivity estimation estimates a selectivity of 0 because the algorithm assumes that the first value in the histogram is the minimum value of the column. For this reason, the better filter k1=k2.t1 is not preferred.

The fix for this issue is change the statistics collection algorithm to include the minimum value in the histogram. 

I have been able to verify this fix by modifying the histogram manually and adding value 1 in the beginning. With this I was able to get the expected query plan.
